src/main/java/com/bookstore/service/category/CategoryServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+package com.bookstore.service.category;
+
+import com.bookstore.dto.category.CategoryDto;
+import com.bookstore.dto.category.CreateCategoryRequestDto;
+import com.bookstore.entity.Category;
+import com.bookstore.mapper.CategoryMapper;
+import com.bookstore.repository.CategoryRepository;
+import jakarta.persistence.EntityNotFoundException;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll(pageable).stream()
+                .map(categoryMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryDto findById(Long id) {
+        Category category = categoryRepository.findById(id)
+                .orElseThrow(EntityNotFoundException::new);
+        return categoryMapper.toDto(category);
+    }
+
+    @Override
+    public CategoryDto save(CreateCategoryRequestDto categoryDto) {
+        return categoryMapper.toDto(categoryRepository.save(categoryMapper.toEntity(categoryDto)));
```

Comment:
let's improve readability
```suggestion
        Category category = categoryMapper.toEntity(categoryDto);
        return categoryMapper.toDto(categoryRepository.save(category));
```